pinot-core/src/main/java/org/apache/pinot/core/operator/combine/SelectionOnlyCombineOperator.java

```diff
@@ -60,7 +60,7 @@ protected IntermediateResultsBlock getNextBlock() {
     // For LIMIT 0 query, only process one segment to get the data schema
     if (_numRowsToKeep == 0) {
       IntermediateResultsBlock resultsBlock = (IntermediateResultsBlock) _operators.get(0).nextBlock();
-      CombineOperatorUtils.setExecutionStatistics(resultsBlock, _operators);
+      CombineOperatorUtils.setExecutionStatistics(resultsBlock, _operators, 0);
```

Comment:
0 is not good. LIMIT 0 queries should be rare but we still process the query on one segment. So, I think we should instrument it and add the actual value here since this is a special case.

Reply:
It does not use the worker thread at all, so IMO it is okay to put 0 here. It won't actually process the segment, but just fetch the schema.

Reply:
Yes, we don't need to time the execution time here since this is a single thread case. For all single thread execution time, timing is conducted at `InstanceResponseBlock.getNetxBlock()`

Reply:
Ohh yes. Correct.